docs/tutorials/agent_framework/Chatbot_with_RAG.md

```diff
@@ -0,0 +1,318 @@
+{"_index":"test_tech_news","_source":{"passage":"Amazon Bedrock is a fully managed service that offers a choice of high-performing foundation models (FMs) from leading AI companies like AI21 Labs, Anthropic, Cohere, Meta, Stability AI, and Amazon via a single API, along with a broad set of capabilities you need to build generative AI applications with security, privacy, and responsible AI. Using Amazon Bedrock, you can easily experiment with and evaluate top FMs for your use case, privately customize them with your data using techniques such as fine-tuning and Retrieval Augmented Generation (RAG), and build agents that execute tasks using your enterprise systems and data sources. Since Amazon Bedrock is serverless, you don\u0027t have to manage any infrastructure, and you can securely integrate and deploy generative AI capabilities into your applications using the AWS services you are already familiar with."},"_id":"mLU8xI0B8vrNLhb9yBpV","_score":0.5604863}
+"""
+        },
+        {
+          "name": "response",
+          "result": "Vision Pro is a mixed-reality headset developed by Apple that was announced in 2023. It uses cameras and sensors to overlay digital objects and information on the real world. The device runs an operating system called visionOS that allows users to interact with windows and apps in a 3D environment using gestures, eye tracking, and voice commands."
+        }
+      ]
+    }
+  ]
+}
+```
+You can also check detail steps in trace data:
+```
+GET _plugins/_ml/memory/message/ebVSxI0B8vrNLhb9nxty/traces
+```
+
+- Example2: ask question related to city population
+```
+POST _plugins/_ml/agents/your_agent_id/_execute
+{
+  "parameters": {
+    "question": "What's the population of Seattle 2023",
+    "verbose": true
+  }
+}
+```
+From response, we can see Agent runs `population_data_knowledge_base` tool and get top 3 documents. Agent will pass these documents as context to LLM.
```

Comment:
From the response, we can see the Agent runs the `population_data_knowledge_base` tool and retrieves the top 3 documents. The Agent then passes these documents as context to the LLM, which allows it to accurately answer the question.